autorest/azure/azure.go

```diff
@@ -29,12 +29,13 @@ const (
 
 // ServiceError encapsulates the error response from an Azure service.
 type ServiceError struct {
-	Code    string `json:"code"`
-	Message string `json:"message"`
+	Code    string       `json:"code"`
+	Message string       `json:"message"`
+	Details *interface{} `json:"details"`
 }
 
 func (se ServiceError) Error() string {
-	return fmt.Sprintf("Azure Error: Code=%q Message=%q", se.Code, se.Message)
+	return fmt.Sprintf("Code=%q Message=%q Details=%v", se.Code, se.Message, se.Details)
```

Comment:
I'd love to see a test validating actual output of this method (especially the Details=%v part).

Also I think we should not add `Details=<nil>` if `Details == nil`. It just looks bad. an if statement here would be preferable to prevent adding that unnecessary suffix.